config/stp.py

```diff
@@ -331,45 +505,92 @@ def spanning_tree(_db):
-    # Enable STP for VLAN by default
-    enable_stp_for_interfaces(db)
-    enable_stp_for_vlans(db)
+    elif mode == "mst" and current_mode == "mst":
+        ctx.fail("MST is already configured")
+
+    if mode == "pvst":
+        disable_global_mst(db)
+
+        fvs = {'mode': mode,
+            'rootguard_timeout': STP_DEFAULT_ROOT_GUARD_TIMEOUT,
+            'forward_delay': STP_DEFAULT_FORWARD_DELAY,
+            'hello_time': STP_DEFAULT_HELLO_INTERVAL,
+            'max_age': STP_DEFAULT_MAX_AGE,
+            'priority': STP_DEFAULT_BRIDGE_PRIORITY
+            }
+        db.set_entry('STP', "GLOBAL", fvs)
+        
+        enable_stp_for_interfaces(db)
+        enable_stp_for_vlans(db) # Enable STP for VLAN by default
+    
+    elif mode == "mst":
+        disable_global_pvst(db)
+
+        fvs = {'mode': mode,
+               }
+        db.mod_entry('STP', "GLOBAL", fvs)
+
+        bridge_mac = get_bridge_mac_address(db)
+        if not bridge_mac:
+            ctx.fail("Bridge MAC address not found in DEVICE_METADATA table")
+
+        # Setting MSTP parameters in the STP_MST_TABLE
+        mst_fvs = {
+            'name': bridge_mac,
+            'revision': MST_DEFAULT_REVISION,
+            'max_hop': MST_DEFAULT_HOPS,
+            'max_age': MST_DEFAULT_MAX_AGE,
+            'hello_time': MST_DEFAULT_HELLO_INTERVAL,
+            'forward_delay': MST_DEFAULT_FORWARD_DELAY
+        }
+        db.set_entry('STP_MST', "GLOBAL", mst_fvs)
+
+        do_vlan_to_instance0(db) # VLANs to Instance 0 mapping as part of global configuration
+        enable_mst_for_interfaces(db)
 
 
 # cmd: STP disable
+# config spanning_tree disable <pvst|mst> (Modify mode parameter for MST or PVST and Delete tables)
+# Modify mode in STP GLOBAL table to None
+# Delete tables STP_MST, STP_MST_INST, STP_MST_PORT, and STP_PORT
 @spanning_tree.command('disable')
-@click.argument('mode', metavar='<pvst>', required=True, type=click.Choice(["pvst"]))
+@click.argument('mode', metavar='<pvst|mst>', required=True, type=click.Choice(["pvst","mst"]))
 @clicommon.pass_db
 def stp_disable(_db, mode):
     """disable STP """
+    ctx = click.get_current_context()
     db = _db.cfgdb
-    db.set_entry('STP', "GLOBAL", None)
-    # Disable STP for all VLANs and interfaces
-    db.delete_table('STP_VLAN')
-    db.delete_table('STP_PORT')
-    db.delete_table('STP_VLAN_PORT')
-    if get_global_stp_mode(db) == "pvst":
-        print("Error PVST disable failed")
+    current_mode = get_global_stp_mode(db)
+
+    if not current_mode or current_mode == "none":
+        ctx.fail("STP is not configured")
+    elif mode != current_mode and current_mode!= "none":
+        ctx.fail(f"{mode.upper()} is not currently configured mode")
+
+    if mode == "pvst" and current_mode == "pvst":
+        disable_global_pvst(db)
+        print("PVST is disabled")
+    elif mode == "mst" and current_mode == "mst":
+        disable_global_mst(db)
+        print("MST is disabled")        
```

Comment:
Please remove the print

Reply:
I have addressed this comment, please review